Assets/SEE/DataModel/DG/SourceRange/SourceRange.cs

```diff
@@ -0,0 +1,40 @@
+﻿using UnityEngine.Assertions;
+
+namespace SEE.DataModel.DG.SourceRange
+{
+    /// <summary>
+    /// A representation of a source-code range in the index.
+    /// </summary>
+    internal class SourceRange : FileRanges
```

Comment:
`Range` also extends `FileRanges`. Is this intended?

Reply:
I assume you mean `SourceRange` and not `Range`, which has been done as part of a previous PR. I'm not sure myself why this extends `FileRange`.

Reply:
I'm just wondering why both classes extend `FileRanges`.

Reply:
It's a bit confusing because the former `Range` has been renamed to `SourceRange` in this pull request. I can explain why `SourceRange` (formerly known as `Range` inherits from `FileRanges`. In the following I am using the term `SourceRange` for the class that existed prior to this pull request and was formerly known as `Range`, while the use of `Range` refers to the newly introduced class.

The `SourceRangeIndex` is a two-level structure indexed by a path. The first level consists of `FileRanges`, which consist of nested `SourceRange`s (second level). A `FileRanges` has no source position (start line, end line) and is not associated with a `Node.`  It is just a container for all nested `SourceRange`s, which do have the latter properties and, since they are nested, may also have `Children`.  `FileRanges` and `SourceRange`s are things that are part of a `SourceRangeIndex` and they share the concept of `Children` of type `SourceRange`. That is why `SourceRange` derives from `FileRanges`.

I actually thought about introducing a common abstract superclass for the two of them and have them rather as siblings. Yet, they should be considered implementation details of `SourceRangeIndex` and adding yet another class just adds more code. So I refrained from that.

Having said that, `SourceRange` and `FileRanges` are implementation details of `SourceRangeIndex` and intended as structures contained therein. Apparently the former name `Range` collided with the newly concept for a represenation of source-code regions in general, which has led to the renaming from the former `Range` to `SourceRange` and the inheritance of the new `Range` from `FileRanges`.  `Range` now has `Children` of type `SourceRange`, too. I don't think that was intentional.

That is why the question by @msteinbeck is valid. The inheritance of `Range` should be removed and `Range` should not appear in the namespace `SourceRange`. To avoid future confusion, we should rename the namespace `SourceRange` to `GraphIndex` or similar (there is a already another namespace in `SEE.DataModel,DG.GraphSearch` which serves a similar purpose: searching for nodes; so a sibling namespace `GraphIndex` would fit).

Reply:
> inheritance of the new Range from FileRanges. Range now has Children of type SourceRange, too. I don't think that was intentional.

The new `Range` class (or rather, record) does not inherit from `FileRanges`. It may look like that on first glance since the diff is a little confusing (as you mentioned, previously `SourceRange` was named `Range`).

I agree that `Range` should be in another namespace. I've moved it to `SEE.DataModel.DG` and renamed the `SourceRange` namespace to `GraphIndex` accordingly. Please check whether the new structure is fine.

Reply:
Fine now.